
Remove faulty sed pattern for mingw sed 3.0.2.

---

diff -ur tmp/libiconv-1.12/windows/windres-options src/libiconv-1.12/windows/windres-options
--- src/libiconv-1.12/windows/windres-options 2007-05-27 19:42:10 +0100
+++ libiconv-1.12/windows/windres-options 2008-06-30 15:27:31 +0100
@@ -13,28 +13,11 @@
   shift
 fi
 version="$1" # something like 2.0 or 2.17 or 2.17.3 or 2.17.3-pre3
-
-sed_extract_major='/^[0-9]/{s/^\([0-9]*\).*/\1/p;q}
-a\
-0
-q
-'
-sed_extract_minor='/^[0-9][0-9]*[.][0-9]/{s/^[0-9]*[.]\([0-9]*\).*/\1/p;q}
-a\
-0
-q
-'
-sed_extract_subminor='/^[0-9][0-9]*[.][0-9][0-9]*[.][0-9]/{s/^[0-9]*[.][0-9]*[.]\([0-9]*\).*/\1/p;q}
-a\
-0
-q
-'
-
 {
   echo "-DPACKAGE_VERSION_STRING=\"${version}\""
-  echo "-DPACKAGE_VERSION_MAJOR="`echo "${version}" | sed -n -e "$sed_extract_major"`
-  echo "-DPACKAGE_VERSION_MINOR="`echo "${version}" | sed -n -e "$sed_extract_minor"`
-  echo "-DPACKAGE_VERSION_SUBMINOR="`echo "${version}" | sed -n -e "$sed_extract_subminor"`
+  echo "-DPACKAGE_VERSION_MAJOR=`IFS=.; set ${version}; echo ${1-0}`"
+  echo "-DPACKAGE_VERSION_MINOR=`IFS=.; set ${version}; echo ${2-0}`"
+  echo "-DPACKAGE_VERSION_SUBMINOR=`IFS=.; set ${version}; echo ${3-0}`"
 } |
 {
   if test -n "$escape"; then
