
Remove faulty sed pattern for mingw sed 3.0.2.

---

--- a/windows/windres-options	2009-06-21 14:17:33.000000000 +0300
+++ b/windows/windres-options	2009-10-10 02:19:57.457105218 +0200
@@ -32,9 +32,9 @@ q
 
 {
   echo "-DPACKAGE_VERSION_STRING=\"${version}\""
-  echo "-DPACKAGE_VERSION_MAJOR="`echo "${version}" | sed -n -e "$sed_extract_major"`
-  echo "-DPACKAGE_VERSION_MINOR="`echo "${version}" | sed -n -e "$sed_extract_minor"`
-  echo "-DPACKAGE_VERSION_SUBMINOR="`echo "${version}" | sed -n -e "$sed_extract_subminor"`
+  echo "-DPACKAGE_VERSION_MAJOR=`IFS=.; set ${version}; echo ${1-0}`"
+  echo "-DPACKAGE_VERSION_MINOR=`IFS=.; set ${version}; echo ${2-0}`"
+  echo "-DPACKAGE_VERSION_SUBMINOR=`IFS=.; set ${version}; echo ${3-0}`"
 } |
 {
   if test -n "$escape"; then
