Index: a/src/libopensc/internal-winscard.h
===================================================================
--- a/src/libopensc/internal-winscard.h	(revision 3651)
+++ b/src/libopensc/internal-winscard.h	(working copy)
@@ -128,8 +128,13 @@
 /* Copied from pcsc-lite reader.h */
 
 #ifndef SCARD_CTL_CODE
+#ifdef _WIN32
+#include <winioctl.h>
+#define SCARD_CTL_CODE(code) CTL_CODE(FILE_DEVICE_SMARTCARD,(code),METHOD_BUFFERED,FILE_ANY_ACCESS)
+#else
 #define SCARD_CTL_CODE(code) (0x42000000 + (code))
 #endif
+#endif
 
 /**
  * TeleTrust Class 2 reader tags
