Fixed in HEAD

---

diff -urNp openssl-1.0.0-beta4.org/crypto/rand/rand.h openssl-1.0.0-beta4/crypto/rand/rand.h
--- openssl-1.0.0-beta4.org/crypto/rand/rand.h	2008-11-12 05:58:04.000000000 +0200
+++ openssl-1.0.0-beta4/crypto/rand/rand.h	2010-01-11 17:13:15.796020568 +0200
@@ -63,10 +63,6 @@
 #include <openssl/ossl_typ.h>
 #include <openssl/e_os2.h>
 
-#if defined(OPENSSL_SYS_WINDOWS)
-#include <windows.h>
-#endif
-
 #ifdef  __cplusplus
 extern "C" {
 #endif
diff -urNp openssl-1.0.0-beta4.org/e_os.h openssl-1.0.0-beta4/e_os.h
--- openssl-1.0.0-beta4.org/e_os.h	2009-08-26 18:13:43.000000000 +0300
+++ openssl-1.0.0-beta4/e_os.h	2010-01-11 17:13:08.152360026 +0200
@@ -372,6 +372,13 @@ static unsigned int _strlen31(const char
 #    define DEFAULT_HOME  "C:"
 #  endif
 
+#undef X509_NAME
+#undef X509_EXTENSIONS
+#undef X509_CERT_PAIR
+#undef PKCS7_ISSUER_AND_SERIAL
+#undef OCSP_REQUEST
+#undef OCSP_RESPONSE
+
 #else /* The non-microsoft world */
 
 #  ifdef OPENSSL_SYS_VMS
