--- openssl-SNAP-20081104/crypto/bn/bn_asm.c	2008-11-02 21:00:07.000000000 +0200
+++ openssl-SNAP-20081104.new/crypto/bn/bn_asm.c	2008-11-05 21:28:48.000000000 +0200
@@ -68,7 +68,7 @@
 
 #if defined(BN_LLONG) || defined(BN_UMULT_HIGH)
 
-BN_ULONG bn_mul_add_words(BN_ULONG *rp, const BN_ULONG *ap, int num, BN_ULONG w)
+BN_ULONG bn_mul_add_words(BN_ULONG *rp, const BN_ULONG *ap, size_t num, BN_ULONG w)
 	{
 	BN_ULONG c1=0;
 
@@ -94,7 +94,7 @@ BN_ULONG bn_mul_add_words(BN_ULONG *rp, 
 	return(c1);
 	} 
 
-BN_ULONG bn_mul_words(BN_ULONG *rp, const BN_ULONG *ap, int num, BN_ULONG w)
+BN_ULONG bn_mul_words(BN_ULONG *rp, const BN_ULONG *ap, size_t num, BN_ULONG w)
 	{
 	BN_ULONG c1=0;
 
@@ -119,7 +119,7 @@ BN_ULONG bn_mul_words(BN_ULONG *rp, cons
 	return(c1);
 	} 
 
-void bn_sqr_words(BN_ULONG *r, const BN_ULONG *a, int n)
+void bn_sqr_words(BN_ULONG *r, const BN_ULONG *a, size_t n)
         {
 	assert(n >= 0);
 	if (n <= 0) return;
--- openssl-SNAP-20081104/crypto/rand/rand_egd.c	2008-11-01 19:00:11.000000000 +0200
+++ openssl-SNAP-20081104.new/crypto/rand/rand_egd.c	2008-11-05 21:30:46.000000000 +0200
@@ -96,7 +96,7 @@
  */
 
 #if defined(OPENSSL_SYS_WIN32) || defined(OPENSSL_SYS_VMS) || defined(OPENSSL_SYS_MSDOS) || defined(OPENSSL_SYS_VXWORKS) || defined(OPENSSL_SYS_NETWARE) || defined(OPENSSL_SYS_VOS) || defined(OPENSSL_SYS_BEOS)
-int RAND_query_egd_bytes(const char *path, unsigned char *buf, int bytes)
+int RAND_query_egd_bytes(const char *path, unsigned char *buf, size_t bytes)
 	{
 	return(-1);
 	}
@@ -105,7 +105,7 @@ int RAND_egd(const char *path)
 	return(-1);
 	}
 
-int RAND_egd_bytes(const char *path,int bytes)
+int RAND_egd_bytes(const char *path,size_t bytes)
 	{
 	return(-1);
 	}
