
Support cross compile.

---

diff -urNp zlib-1.2.3.org/win32/Makefile.gcc zlib-1.2.3/win32/Makefile.gcc
--- zlib-1.2.3.org/win32/Makefile.gcc	2003-08-02 06:49:00.000000000 +0300
+++ zlib-1.2.3/win32/Makefile.gcc	2008-10-20 11:58:58.000000000 +0200
@@ -30,7 +30,8 @@ IMPLIB    = libzdll.a
 #LOC = -DASMV
 #LOC = -DDEBUG -g
 
-CC = gcc
+PREFIX =
+CC = $(PREFIX)gcc
 CFLAGS = $(LOC) -O3 -Wall
 
 AS = $(CC)
@@ -39,10 +40,10 @@ ASFLAGS = $(LOC) -Wall
 LD = $(CC)
 LDFLAGS = $(LOC) -s
 
-AR = ar
+AR = $(PREFIX)ar
 ARFLAGS = rcs
 
-RC = windres
+RC = $(PREFIX)windres
 RCFLAGS = --define GCC_WINDRES
 
 CP = cp -fp
@@ -79,9 +80,9 @@ $(STATICLIB): $(OBJS) $(OBJA)
 $(IMPLIB): $(SHAREDLIB)
 
 $(SHAREDLIB): win32/zlib.def $(OBJS) $(OBJA) zlibrc.o
-	dllwrap --driver-name $(CC) --def win32/zlib.def \
+	$(PREFIX)dllwrap --driver-name $(CC) --def win32/zlib.def \
 	  --implib $(IMPLIB) -o $@ $(OBJS) $(OBJA) zlibrc.o
-	strip $@
+	$(PREFIX)strip $@
 
 example: example.o $(STATICLIB)
 	$(LD) $(LDFLAGS) -o $@ example.o $(STATICLIB)
@@ -99,23 +100,26 @@ zlibrc.o: win32/zlib1.rc
 	$(RC) $(RCFLAGS) -o $@ win32/zlib1.rc
 
 
-# INCLUDE_PATH and LIBRARY_PATH must be set.
+# INCLUDE_PATH, LIBRARY_PATH and BININARY_PATH must be set.
 
 .PHONY: install uninstall clean
 
-install: zlib.h zconf.h $(LIB)
-	-@if not exist $(INCLUDE_PATH)/nul mkdir $(INCLUDE_PATH)
-	-@if not exist $(LIBRARY_PATH)/nul mkdir $(LIBRARY_PATH)
+install: zlib.h zconf.h $(STATICLIB) $(IMPLIB)
+	-mkdir -p $(INCLUDE_PATH)
+	-mkdir -p $(LIBRARY_PATH)
+	-mkdir -p $(BINARY_PATH)
 	-$(INSTALL) zlib.h $(INCLUDE_PATH)
 	-$(INSTALL) zconf.h $(INCLUDE_PATH)
 	-$(INSTALL) $(STATICLIB) $(LIBRARY_PATH)
 	-$(INSTALL) $(IMPLIB) $(LIBRARY_PATH)
+	-$(INSTALL) $(SHAREDLIB) $(BINARY_PATH)
 
 uninstall:
 	-$(RM) $(INCLUDE_PATH)/zlib.h
 	-$(RM) $(INCLUDE_PATH)/zconf.h
 	-$(RM) $(LIBRARY_PATH)/$(STATICLIB)
 	-$(RM) $(LIBRARY_PATH)/$(IMPLIB)
+	-$(RM) $(BINARY_PATH)/$(SHAREDLIB)
 
 clean:
 	-$(RM) $(STATICLIB)
