diff -urNp zlib-1.2.4.org/win32/Makefile.gcc zlib-1.2.4/win32/Makefile.gcc
--- zlib-1.2.4.org/win32/Makefile.gcc	2010-02-23 18:00:23.000000000 +0200
+++ zlib-1.2.4/win32/Makefile.gcc	2010-03-17 15:43:28.332296998 +0200
@@ -30,7 +30,8 @@ IMPLIB    = libzdll.a
 #LOC = -DASMV
 #LOC = -DDEBUG -g
 
-CC = gcc
+PREFIX =
+CC = $(PREFIX)gcc
 CFLAGS = $(LOC) -O3 -Wall
 
 AS = $(CC)
@@ -39,13 +40,13 @@ ASFLAGS = $(LOC) -Wall
 LD = $(CC)
 LDFLAGS = $(LOC) -s
 
-AR = ar
+AR = $(PREFIX)ar
 ARFLAGS = rcs
 
-RC = windres
+RC = $(PREFIX)windres
 RCFLAGS = --define GCC_WINDRES
 
-STRIP = strip
+STRIP = $(PREFIX)strip
 
 CP = cp -fp
 # If GNU install is available, replace $(CP) with install.
@@ -101,23 +102,26 @@ zlibrc.o: win32/zlib1.rc
 	$(RC) $(RCFLAGS) -o $@ win32/zlib1.rc
 
 
-# INCLUDE_PATH and LIBRARY_PATH must be set.
+# BINARY_PATH, INCLUDE_PATH and LIBRARY_PATH must be set.
 
 .PHONY: install uninstall clean
 
-install: zlib.h zconf.h $(LIB)
+install: all zlib.h zconf.h $(LIB)
-	-@if not exist $(INCLUDE_PATH)/nul mkdir $(INCLUDE_PATH)
-	-@if not exist $(LIBRARY_PATH)/nul mkdir $(LIBRARY_PATH)
+	-mkdir -p $(INCLUDE_PATH)
+	-mkdir -p $(LIBRARY_PATH)
+	-mkdir -p $(BINARY_PATH)
 	-$(INSTALL) zlib.h $(INCLUDE_PATH)
 	-$(INSTALL) zconf.h $(INCLUDE_PATH)
 	-$(INSTALL) $(STATICLIB) $(LIBRARY_PATH)
 	-$(INSTALL) $(IMPLIB) $(LIBRARY_PATH)
+	-$(INSTALL) $(SHAREDLIB) $(BINARY_PATH)
 
 uninstall:
 	-$(RM) $(INCLUDE_PATH)/zlib.h
 	-$(RM) $(INCLUDE_PATH)/zconf.h
 	-$(RM) $(LIBRARY_PATH)/$(STATICLIB)
 	-$(RM) $(LIBRARY_PATH)/$(IMPLIB)
+	-$(RM) $(BINARY_PATH)/$(SHAREDLIB)
 
 clean:
 	-$(RM) $(STATICLIB)
