diff -urNp zlib-1.2.4.org/configure zlib-1.2.4/configure
--- zlib-1.2.4.org/configure	2010-03-12 17:06:14.000000000 +0200
+++ zlib-1.2.4/configure	2010-03-17 16:23:47.556136194 +0200
@@ -13,15 +13,21 @@
 # If you have problems, try without defining CC and CFLAGS before reporting
 # an error.
 
+if [ -n "${CHOST}" ]; then
+	uname="$(echo "${CHOST}" | sed 's/.*-.*-\(.*\)-.*/\1/')"
+	CROSS_PREFIX="${CHOST}-"
+fi
+
 STATICLIB=libz.a
 LDFLAGS="${LDFLAGS} -L. ${STATICLIB}"
 VER=`sed -n -e '/VERSION "/s/.*"\(.*\)".*/\1/p' < zlib.h`
 VER3=`sed -n -e '/VERSION "/s/.*"\([0-9]*\\.[0-9]*\\.[0-9]*\).*/\1/p' < zlib.h`
 VER2=`sed -n -e '/VERSION "/s/.*"\([0-9]*\\.[0-9]*\)\\..*/\1/p' < zlib.h`
 VER1=`sed -n -e '/VERSION "/s/.*"\([0-9]*\)\\..*/\1/p' < zlib.h`
-AR=${AR-"ar"}
-AR_RC="${AR} rc"
-RANLIB=${RANLIB-"ranlib"}
+AR=${AR-"${CROSS_PREFIX}ar"}
+AR_RC=${AR_RC-"${CROSS_PREFIX}ar rc"}
+RANLIB=${RANLIB-"${CROSS_PREFIX}ranlib"}
+LDCONFIG=${LDCONFIG-"ldconfig"}
 prefix=${prefix-/usr/local}
 exec_prefix=${exec_prefix-'${prefix}'}
 libdir=${libdir-'${exec_prefix}/lib'}
@@ -69,7 +75,7 @@ int hello() {return getchar();}
 EOF
 
 test -z "$CC" && echo Checking for gcc...
-cc=${CC-gcc}
+cc=${CC-${CROSS_PREFIX}gcc}
 cflags=${CFLAGS-"-O3"}
 # to force the asm version use: CFLAGS="-O3 -DASMV" ./configure
 case "$cc" in
@@ -91,7 +97,7 @@ if test "$gcc" -eq 1 && ($cc -c $cflags 
     uname=`(uname -s || echo unknown) 2>/dev/null`
   fi
   case "$uname" in
-  Linux | linux | GNU | GNU/* | *BSD | DragonFly) LDSHARED=${LDSHARED-"$cc -shared -Wl,-soname,libz.so.1,--version-script,zlib.map"} ;;
+  Linux* | linux* | GNU | GNU/* | *BSD | DragonFly) LDSHARED=${LDSHARED-"$cc -shared -Wl,-soname,libz.so.1,--version-script,zlib.map"} ;;
   CYGWIN* | Cygwin* | cygwin* | OS/2* )
              EXE='.exe' ;;
   QNX*)  # This is for QNX6. I suppose that the QNX rule below is for QNX2,QNX4
@@ -513,6 +519,7 @@ sed < Makefile.in "
 /^SHAREDLIBM *=/s#=.*#=$SHAREDLIBM#
 /^AR *=/s#=.*#=$AR_RC#
 /^RANLIB *=/s#=.*#=$RANLIB#
+/^LDCONFIG *=/s#=.*#=$LDCONFIG#
 /^EXE *=/s#=.*#=$EXE#
 /^prefix *=/s#=.*#=$prefix#
 /^exec_prefix *=/s#=.*#=$exec_prefix#
diff -urNp zlib-1.2.4.org/Makefile.in zlib-1.2.4/Makefile.in
--- zlib-1.2.4.org/Makefile.in	2010-03-14 04:10:27.000000000 +0200
+++ zlib-1.2.4/Makefile.in	2010-03-17 16:03:32.182849536 +0200
@@ -16,6 +16,7 @@
 # To install in $HOME instead of /usr/local, use:
 #    make install prefix=$HOME
 
+LDCONFIG=ldconfig
 CC=cc
 
 CFLAGS=-O
@@ -171,7 +172,7 @@ install-libs: $(LIBS)
 	  rm -f $(SHAREDLIB) $(SHAREDLIBM); \
 	  ln -s $(SHAREDLIBV) $(SHAREDLIB); \
 	  ln -s $(SHAREDLIBV) $(SHAREDLIBM); \
-	  (ldconfig || true)  >/dev/null 2>&1; \
+	  ($(LDCONFIG) || true)  >/dev/null 2>&1; \
 	fi
 	cp zlib.3 $(DESTDIR)$(man3dir)
 	chmod 644 $(DESTDIR)$(man3dir)/zlib.3
